app/src/main/java/org/cru/godtools/ui/tooldetails/ToolDetailsFragment.kt

```diff
@@ -76,6 +78,22 @@ class ToolDetailsFragment() :
         menu.setupPinShortcutAction()
     }
 
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        when (requestCode) {
+            REQUEST_TUTORIAL_TIPS -> when (resultCode) {
+                Activity.RESULT_OK -> {
+                    settings.setFeatureDiscovered("$FEATURE_TUTORIAL_TIPS${dataModel.tipsTool}")
```

Comment:
let's move this into `launchTrainingTips`, set the feature discovered when it is taking the branch that is launching the tract and not the tutorial

Reply:
So something like this?
```
if (force || settings.isFeatureDiscovered("$FEATURE_TUTORIAL_TIPS$code")) {
     settings.setFeatureDiscovered("$FEATURE_TUTORIAL_TIPS${dataModel.tipsTool}")
     requireActivity().openToolActivity(code, type, locale, showTips = true)
}
```
Or should I have a specific case for just ```force``` being true and a case for when ```settings.isFeatureDiscovered("$FEATURE_TUTORIAL_TIPS$code")``` is true?

Reply:
your example is what I'd expect, no special case needed for `force`.

Reply:
also, if you want you can rename `force` to `skipTutorial`, it would make the concept of the flag more straight-forward

Reply:
Yeah I like ```skipTutorial``` better, will change